pkgs/development/python-modules/keymap-drawer/default.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+
+  nix-update-script,
+  pcpp,
+  platformdirs,
+  poetry-core,
+  pydantic,
+  pydantic-settings,
+  pyparsing,
+  pyyaml,
+  stdenv,
+  tree-sitter,
+  tree-sitter-grammars,
+  tree-sitter-native,
+  versionCheckHook,
+}:
+let
+  version = "0.22.0";
+  owner = "caksoylar";
+  repo = "keymap-drawer";
+in
+buildPythonPackage {
+  inherit version;
+  pname = "keymap-drawer";
+  pyproject = true;
+  disabled = pythonOlder "3.12";
+
+  src = fetchFromGitHub {
+    inherit owner repo;
+    tag = "v${version}";
+    hash = "sha256-SPnIfrUA0M9xznjEe60T+0VHh9lCmY4cni9hyqFlZqM=";
+  };
+
+  build-system = [ poetry-core ];
+
+  dependencies = [
+    pcpp
+    platformdirs
+    pydantic
+    pydantic-settings
+    pyparsing
+    pyyaml
+    tree-sitter
+    tree-sitter-grammars.tree-sitter-devicetree
+  ];
+
+  nativeCheckInputs = [
+    versionCheckHook
+  ];
+
+  pythonImportsCheck = [ "keymap_drawer" ];
+
+  versionCheckProgram = "${placeholder "out"}/bin/keymap";
+  versionCheckProgramArg = "--version";
+  versionCheckKeepEnvironment = [ "DYLD_LIBRARY_PATH" ];
+
+  env = lib.optionalAttrs stdenv.buildPlatform.isDarwin {
+    DYLD_LIBRARY_PATH = tree-sitter-native.builtGrammars.tree-sitter-devicetree;
+  };
```

setting things in `env` only affect build time environment, we want to wrap the program so the dynamic linker can find the library at runtime

`LD_LIBRARY_PATH` is probably set by default, that might why it works on linux

```suggestion
  makeWrapperArgs = [
    "--prefix"
    (if stdenv.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH")
    ":"
    tree-sitter-native.builtGrammars.tree-sitter-devicetree
  ];
```

---

i think it’s possible to fix this up at python tree sitter

will take a look the the coming week

---

cooked up a fix here #420196
it's less difficult than i imagined